src/main/java/com/exasol/containers/ExasolContainer.java

```diff
@@ -268,28 +270,54 @@ public T withClusterLogsPath(final Path clusterLogsHostPath) {
         return self();
     }
 
+    @Override
+    protected void containerIsStarting(final InspectContainerResponse containerInfo, final boolean reused) {
+        super.containerIsStarting(containerInfo, reused);
+        this.reused = reused;
+    }
+
     // [impl->dsn~exasol-container-ready-criteria~3]
     @Override
     protected void waitUntilContainerStarted() {
         waitUntilClusterConfigurationAvailable();
         final Instant afterUtc = Instant.now();
         waitUntilStatementCanBeExecuted();
         if (this.requiredServices.contains(ExasolService.BUCKETFS)) {
-            new BucketFsWaitStrategy(this.detectorFactory, afterUtc).waitUntilReady(this);
+            if (!this.reused) {
+                new BucketFsWaitStrategy(this.detectorFactory, afterUtc).waitUntilReady(this);
+            }
             this.readyServices.add(ExasolService.BUCKETFS);
         }
         if (this.requiredServices.contains(ExasolService.UDF)) {
-            new UdfContainerWaitStrategy(this.detectorFactory, afterUtc).waitUntilReady(this);
+            if (!this.reused) {
+                new UdfContainerWaitStrategy(this.detectorFactory, afterUtc).waitUntilReady(this);
+            }
             this.readyServices.add(ExasolService.UDF);
         }
         logger().info("Exasol container started after waiting for the following services to become available: {}",
                 this.requiredServices);
+        if (this.reused) {
+            purgeDatabase();
+        }
+    }
+
+    // [impl->dsn~purging~1]
```

Comment:
Okay, this again goes to previous comment if purging can be a feature on its own.
I am not sure if private api/method can be a design item. --> ask @redcatbear